src/api/config.rs

```diff
@@ -508,6 +508,27 @@ impl SpeedSettings {
   }
 }
 
+/// Range of block sizes to use.
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+pub struct PartitionRange {
+  pub(crate) min: BlockSize,
+  pub(crate) max: BlockSize,
+}
+
+impl PartitionRange {
+  /// Creates a new partition range with min and max partition sizes.
+  pub fn new(min: BlockSize, max: BlockSize) -> Self {
+    assert!(max >= min);
```

Comment:
You should document what blocks are acceptable and add a validation check.

It can happen on a second set if you want to land this now.